src/Model/test.js

```diff
@@ -716,6 +716,51 @@ describe('Model observation', () => {
     scheduler.expectObservable(c$).toBe(cExpected, { m: model })
     scheduler.flush()
   })
+  it('notifies subscribers of changes and deletion', async () => {
+    const { tasks } = mockDatabase()
+    const task = await tasks.create()
+
+    const observer1 = jest.fn()
+    const unsubscribe1 = task.experimentalSubscribe(observer1)
+    expect(observer1).toHaveBeenCalledTimes(0)
+
+    await task.update()
+    expect(observer1).toHaveBeenCalledTimes(1)
+    expect(observer1).toHaveBeenLastCalledWith(false)
+
+    const observer2 = jest.fn()
+    const unsubscribe2 = task.experimentalSubscribe(observer2)
+    expect(observer2).toHaveBeenCalledTimes(0)
+
+    unsubscribe1()
+
+    const observer3 = jest.fn()
+    const unsubscribe3 = task.experimentalSubscribe(observer3)
+
+    await task.update()
+
+    expect(observer2).toHaveBeenCalledTimes(1)
+    expect(observer3).toHaveBeenCalledTimes(1)
+
+    unsubscribe2()
+
+    await task.update()
+
+    expect(observer3).toHaveBeenCalledTimes(2)
+    expect(observer3).toHaveBeenLastCalledWith(false)
+
+    await task.markAsDeleted()
+
+    expect(observer3).toHaveBeenCalledTimes(3)
+    expect(observer3).toHaveBeenLastCalledWith(true)
+
+    unsubscribe3()
+    unsubscribe3()
```

Comment:
We could also test if calling `unsubscribe3` won't delete other subscribers sth like this:
```
const observer1 = jest.fn()
const unsubscribe1 = task.experimentalSubscribe(observer1)
const observer2 = jest.fn()
const unsubscribe2 = task.experimentalSubscribe(observer2)
const observer3 = jest.fn()
const unsubscribe3 = task.experimentalSubscribe(observer3)

unsubscribe3()
unsubscribe3()
unsubscribe3()

expect(task._subcribers.length == 2)
expect(task._subscribers).toEqual([observer1, observer2])
```

Reply:
I wrote tests that are based on behavior, not checking private properties

Reply:
That's true, you are right. However it could be written like that:
```
const observer1 = jest.fn()
const unsubscribe1 = task.experimentalSubscribe(observer1)
const observer2 = jest.fn()
const unsubscribe2 = task.experimentalSubscribe(observer2)
const observer3 = jest.fn()
const unsubscribe3 = task.experimentalSubscribe(observer3)

unsubscribe3()
unsubscribe3()
unsubscribe3()

await task.update()

expect(observer1).toHaveBeenCalledTimes(1)
expect(observer2).toHaveBeenCalledTimes(1)
expect(observer3).toHaveBeenCalledTimes(0)
```
It's testing a behavior and checks if multiple unsubscribes don't mess other subscriptions.